quickfixj-core/src/main/java/quickfix/mina/WatermarkTracker.java

```diff
@@ -0,0 +1,185 @@
+/*******************************************************************************
+ * Copyright (c) quickfixengine.org  All rights reserved.
+ *
+ * This file is part of the QuickFIX FIX Engine
+ *
+ * This file may be distributed under the terms of the quickfixengine.org
+ * license as defined by quickfixengine.org and appearing in the file
+ * LICENSE included in the packaging of this file.
+ *
+ * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING
+ * THE WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE.
+ *
+ * See http://www.quickfixengine.org/LICENSE for licensing information.
+ *
+ * Contact ask@quickfixengine.org if any conditions of this licensing
+ * are not clear to you.
+ ******************************************************************************/
+
+/*
+ * Copyright (c) 2018 Vladimir Lysyy (mrbald@github)
```

Comment:
I am not sure if we can mix different licenses and copyrights here. Will have to check.

Reply:
Happy to "donate" it to any license and strip this comment. Will include in the next commit.